Editor/AGS.Editor/Panes/FreePanControl.cs

```diff
@@ -0,0 +1,101 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Text;
+using System.Windows.Forms;
+
+namespace AGS.Editor
+{
+    public class FreePanControl : BufferedPanel
+    {
+        private const int WS_EX_COMPOSITED = 0x02000000;
+        private int _lastX, _lastY; //used to prevent useless mouse move events
+        private Point _panGrabPoint = Point.Empty;
+        private Cursor _panCursor = Cursors.Hand; // We need a better hand cursor, using the "link" hand for now
+        public event EventHandler PanGrabbed;
+        public event EventHandler PanReleased;
+        public bool IsPanning { get; private set; }
+
+        // Enabling Composite makes scrolling smoother
+        protected override CreateParams CreateParams
+        {
+            get
+            {
+                CreateParams cp = base.CreateParams;
+                cp.ExStyle |= WS_EX_COMPOSITED;
+                return cp;
+            }
+        }
+
+        public FreePanControl() : base()
```

Comment:
You don't need to a empty constructor

Reply:
Can you elaborate? You mean I can omit all these 4 lines of code of the constructor and it will still work in C# ?

Reply:
Empty constructors are generated automatically unless there's a constructor with parameters.

Reply:
IIRC default constructor is generated automatically unless there are constructors with parameters too.

Reply:
Há, Very cool, didn't knew about it. I will fix this in a previous commit and push force

Reply:
When programming classes the compiler will make a default constructor for you if you haven't explicitly made any. The difference when programming structs is that it will always make a default empty constructor even you make a constructor with parameters, otherwise the line `DateTime dateTime` would be illegal, which it isn't

Reply:
I removed the constructor! :]